### PR TITLE
Mana and NVMe enabled in the info field

### PR DESCRIPTION
**Pull request by kamalca**
These changes were cherry picked from `checkpoint_v3_1`.

Adds information about MANA and NVMe capabilities to the info field of LISA test results.

### CODE REVIEW

File: lisa/sut_orchestrator/azure/platform_.py [hunk 1]
```diff
@@ -732,6 +735,26 @@ def _get_node_information(self, node: Node) -> Dict[str, str]:
...
             node.log.debug(f"vm generation: {information[KEY_VM_GENERATION]}")
+            node.log.debug("detecting mana driver enabled...")
+            information[KEY_MANA_ENABLED] = node.tools[KernelConfig].is_enabled(
+                "CONFIG_MICROSOFT_MANA"
+            )
+            node.log.debug(f"mana enabled: {information[KEY_MANA_ENABLED]}")
+
+            node.log.debug("detecting nvme driver enabled...")
+
+            _has_nvme_core = node.tools[KernelConfig].is_built_in(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py:745]: Should they be test cases, instead of collecting information from all cases?

Comment (kamalca) [reply lisa/sut_orchestrator/azure/platform_.py:745]: I believe we concluded to continue with this approach?

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py:745]: yes

File: lisa/sut_orchestrator/azure/platform_.py [hunk 2]
```diff
@@ -732,6 +735,26 @@ def _get_node_information(self, node: Node) -> Dict[str, str]:
             node.log.debug("detecting vm generation...")
             information[KEY_VM_GENERATION] = node.tools[VmGeneration].get_generation()
             node.log.debug(f"vm generation: {information[KEY_VM_GENERATION]}")
+            node.log.debug("detecting mana driver enabled...")
+            information[KEY_MANA_ENABLED] = node.tools[KernelConfig].is_enabled(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: There should be a method in nic already, please use it.

Comment (kamalca) [reply lisa/sut_orchestrator/azure/platform_.py]: Even inside of nic.py, it looks like this is the way mana driver is checked.
``` python
# handle situation when there is no mana driver, but have mana pci devices
if self._is_mana_device_discovered() and not self._node.tools[
    KernelConfig
].is_enabled("CONFIG_MICROSOFT_MANA"):
```

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py]: I mean it should reuse the method in nic.py. In nic.py, the _is_mana_device_discovered means the mana presents as a pci device. Here the code checks only if mana is in build config. Please double check the logic and try to reuse the nic.py.

Comment (kamalca) [reply lisa/sut_orchestrator/azure/platform_.py]: The goal is to know whether the mana driver is enabled in distros, not to know whether the mana device is actually present. I don't think there is any code in nic.py to reuse unless I created a function just for checking the mana kernel config.

Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py]: So, it needs to be added into nic, and make names clear, including `is_mana_driver_enabled`, `is_mana_device_present`.

File: lisa/nic.py [hunk 3]
```diff
@@ -437,7 +435,7 @@ def _load_nics(self) -> None:
                     self.nics[nic].pci_slot = pci_device.slot
                 break
 
-    def is_mana_present(self) -> bool:
+    def is_mana_device_present(self) -> bool:
```
Comment (squirrelsc) [inline lisa/nic.py:438]: We can change the original `_is_mana_device_discovered` to the new name, and remove this public method. It just call private method, no other logic.

File: lisa/sut_orchestrator/azure/platform_.py [hunk 4]
```diff
@@ -169,6 +170,8 @@
 KEY_WALA_VERSION = "wala_version"
 KEY_WALA_DISTRO_VERSION = "wala_distro"
 KEY_HARDWARE_PLATFORM = "hardware_platform"
+KEY_MANA_ENABLED = "mana_enabled"
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: To be consistent, it should be `mana_driver_enabled`.

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**